src/OrleansSQLUtils/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Orleans;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.MembershipService;
+using OrleansSQLUtils.Configuration;
+
+namespace OrleansSQLUtils
+{
+    public static class IServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Configure DI container with SqlMemebershipTable
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="configureOptions"></param>
+        /// <returns></returns>
+        public static IServiceCollection UseSqlMembershipTable(this IServiceCollection services,
+            Action<SqlMembershipTableOptions> configureOptions)
+        {
+            return services.Configure<SqlMembershipTableOptions>(configureOptions)
+                .AddSingleton<IMembershipTable, SqlMembershipTable>();
+        }
+
+        /// <summary>
+        /// Configure DI container with SqlMemebershipTable
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="configuration"></param>
+        /// <returns></returns>
+        public static IServiceCollection UseSqlMembershipTable(this IServiceCollection services,
+            IConfiguration configuration)
```

Comment:
Why is it IConfiguration, what am I missing?

Reply:
IConfiguration is another main pattern Option team support to configure options. So we provide both choices for  user to configure their provider. What is confusing you?